### ISSUE TITLE
Unable to find the schema at...

### ISSUE THREAD

**Issue by JamieKitson** [corvus-dotnet/Corvus.JsonSchema#410]
I am trying to use the generatejsonschematypes tool against [this JSON schema](https://github.com/theopensystemslab/digital-planning-data-schemas/blob/main/schemas/application.json). However, for virtually all paths as the `--rootPath` parameter I get back _Unable to find the schema at..._ is there something wrong with this schema or is there a limitation of the tool or am I using it incorrectly?

Version: 3.1.0+7a3d554f50ee34dea70e2d95163f9d86d3155298

**Comment by mwadams** [comment]
Your schema contains a `$id` keyword for each of the definitions. This forces each of them into their own scope - so where you use your `$ref` keywords in the form `#/definitions/[whatever]` they are relative to the root of the current scope, not the root.

If you want to use $ids for everything, I recommend using URNs instead of fragments, and then explicitly use the URN. Or eliminate the IDs completely unless you really intend them all to be unique scopes.

For example, here's an example removing the unnecessary `$id`s.

[application.json](https://github.com/user-attachments/files/16765909/application.json)

Then I can generate code like this:

```
generatejsonschematypes.exe --rootPath '#/definitions/Agent' --outputPath ./Model --rootNamespace Feature410 .\application.json
```

**Comment by JamieKitson** [comment]
@mwadams Thanks very much.

**Comment by mwadams** [comment]
Ah! Now, that is a bug that I've just fixed in 4.0.0-preview, and will backport to 3.1 today.

**Comment by mwadams** [comment]
@JamieKitson:

I'm about to push a fix for this issue. It should be available in the next hour or so.

Also, note that if you are getting overlong and complicated names for some of your inline schema, you can use the extension keyword `$corvusTypeName` to set an explicit type name for them!

**Comment by mwadams** [comment]
Closed in 3.1.1 and 4.0.0-preview.9
